packages/kbn-unified-data-table/src/components/data_table_columns.tsx

```diff
@@ -118,7 +118,17 @@ function buildEuiGridColumn({
   headerRowHeight?: number;
   customGridColumnsConfiguration?: CustomGridColumnsConfiguration;
 }) {
-  const dataViewField = dataView.getFieldByName(columnName);
+  const dataViewField = !isPlainRecord
+    ? dataView.getFieldByName(columnName)
+    : new DataViewField({
+        name: columnName,
+        type: columnsMeta?.[columnName]?.type ?? 'unknown',
+        esTypes: columnsMeta?.[columnName]?.esType
+          ? ([columnsMeta[columnName].esType] as string[])
+          : undefined,
+        searchable: true,
+        aggregatable: false,
+      });
```

Comment:
I think we need now a separate util function for creating a DataViewField instance for text-based columns to make sure that UnifiedDataTable and UnifiedFieldList (via src/plugins/discover/public/application/main/components/sidebar/lib/get_field_list.ts) have the same input.

Also we could probably control what field we want to be filterable in a single place.

Reply:
I can do that, but as I explain above I dont want to control it because with casting we can be sure that the filtering will work

Reply:
I think this should be handled by the Discovery team, this way of creating fields for text based languages pre-exist my PR. I don't think it should be handled by this PR.

Reply:
I created an issue to track it https://github.com/elastic/kibana/issues/182646 as it will be better to be handled on a separate PR